### DISCUSSION TITLE
⚠️ Bug - Updating Cart items not stable!

### DISCUSSION THREAD

**Discussion by tarikrital** [swellstores/community#498] [Help] [answered]
Welcome, I use `swell-node` to update a guest cart items but the `put` sometime works other not!

I'm using the following code:
```
    const products = [];
    for (const item of guestCart?.items ?? []) {
      if (item.product_id !== productId) {
        products.push({
          product_id: item.product_id,
          quantity: 1,
        });
      }
    }

    await swell.put(`/carts/{id}`, {
      id: $swellCartId(),
      items: products,
    });
```

**Comment by ericingram** [comment]
Can you describe what you mean by "works" or not? Does the request fail or something else?

**Reply by tarikrital** [reply]
I use it to update items sometime it works: I remove item from the array and I call a get request API after and the list items correct.

But second time it POST data i.e empty array but the get request after returns the old items not empty!

![image](https://github.com/swellstores/community/assets/52907282/7ad29af2-d991-49d9-89c0-08841be2a531)

**Comment by tarikrital** [comment, marked as answer]
Using `$set` fixed the issue but I think it is better to mention it in the sample of the node code.

**Reply by ericingram** [reply]
Glad to know you found a solution. We can update docs or sample code if you can point to where the incorrect info is.

**Reply by tarikrital** [reply]
I don't see what is the benefit of having two ways in the update Cart: `merge` and `override` but it wasn't clear at first UX.

![image](https://github.com/swellstores/community/assets/52907282/162c2ea2-5103-4b03-830c-069079652cd8)

**Reply by ericingram** [reply]
Ah I see. We are aware of some confusion around how array merging works and will be planning to update that behavior in the future. We probably need a guide specifically for merging in the meantime. There's certainly a benefit just depends on what you're trying to accomplish.
